Cannot find type definition file for 'vue/__globalTypes_3.4_false'

### Vue - Official extension or vue-tsc version

2.1.0

### VSCode version

1.92.2

### Vue version

3.4.38

### TypeScript version

5.5.4

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (6) x64 Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz
    Memory: 4.45 GB / 15.80 GB
  Binaries:
    Node: 22.3.0 - ~\AppData\Local\fnm_multishells\16464_1724891209939\node.EXE
    npm: 10.8.2 - ~\AppData\Local\fnm_multishells\16464_1724891209939\npm.CMD
    pnpm: 8.15.6 - ~\AppData\Local\pnpm\pnpm.EXE
    bun: 1.0.33 - ~\.bun\bin\bun.EXE
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.22621.3527
```


### Steps to reproduce

1. Update 'vue-tsc' from 2.0.29 to 2.1.0
2. npm run build
3. it built error, and got the information: 
```
 error TS2688: Cannot find type definition file for 'vue/__globalTypes_3.4_false'.

18 /// <reference types="vue/__globalTypes_3.4_false" />

````

### What is expected?

It should build success just like the version 2.0.29.

### What is actually happening?

There are a lot of errors.

### Link to minimal reproduction

_No response_

### Any additional comments?

_No response_

新建一个项目就会发生。

```
Last login: Thu Aug 29 10:54:30 on ttys022
➜  Downloads pnpm create vue@latest
.../1919cfe0fe8-2aa1                     |   +1 +
.../1919cfe0fe8-2aa1                     | Progress: resolved 1, reused 1, downloaded 0, added 1, done

Vue.js - The Progressive JavaScript Framework

✔ 请输入项目名称： … vue-project
✔ 是否使用 TypeScript 语法？ … 否 / 是
✔ 是否启用 JSX 支持？ … 否 / 是
✔ 是否引入 Vue Router 进行单页面应用开发？ … 否 / 是
✔ 是否引入 Pinia 用于状态管理？ … 否 / 是
✔ 是否引入 Vitest 用于单元测试？ … 否 / 是
✔ 是否要引入一款端到端（End to End）测试工具？ › 不需要
✔ 是否引入 ESLint 用于代码质量检测？ … 否 / 是
✔ 是否引入 Prettier 用于代码格式化？ … 否 / 是
✔ 是否引入 Vue DevTools 7 扩展用于调试? (试验阶段) … 否 / 是

正在初始化项目 /Users/yanhao2023/Downloads/vue-project...

项目初始化完成，可执行以下命令：

  cd vue-project
  pnpm install
  pnpm format
  pnpm dev

➜  Downloads cd vue-project/                                    
➜  vue-project pnpm i                
 WARN  5 deprecated subdependencies found: @humanwhocodes/config-array@0.11.14, @humanwhocodes/object-schema@2.0.3, glob@7.2.3, inflight@1.0.6, rimraf@3.0.2
Packages: +429
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 466, reused 428, downloaded 1, added 429, done
node_modules/.pnpm/vue-demi@0.14.10_vue@3.4.38_typescript@5.4.5_/node_modules/vue-demi: Running postinstall script, done in 68ms

dependencies:
+ pinia 2.2.2
+ vue 3.4.38
+ vue-router 4.4.3

devDependencies:
+ @rushstack/eslint-patch 1.10.4
+ @tsconfig/node20 20.1.4
+ @types/jsdom 21.1.7
+ @types/node 20.16.2 (22.5.1 is available)
+ @vitejs/plugin-vue 5.1.2
+ @vitejs/plugin-vue-jsx 4.0.1
+ @vue/eslint-config-prettier 9.0.0
+ @vue/eslint-config-typescript 13.0.0
+ @vue/test-utils 2.4.6
+ @vue/tsconfig 0.5.1
+ eslint 8.57.0 (9.9.1 is available)
+ eslint-plugin-vue 9.27.0
+ jsdom 24.1.3 (25.0.0 is available)
+ npm-run-all2 6.2.2
+ prettier 3.3.3
+ typescript 5.4.5 (5.5.4 is available)
+ vite 5.4.2
+ vite-plugin-vue-devtools 7.3.9
+ vitest 1.6.0 (2.0.5 is available)
+ vue-tsc 2.1.0

Done in 5.8s
➜  vue-project pnpm outdated
┌───────────────────┬─────────┬────────┐
│ Package           │ Current │ Latest │
├───────────────────┼─────────┼────────┤
│ typescript (dev)  │ 5.4.5   │ 5.5.4  │
├───────────────────┼─────────┼────────┤
│ @types/node (dev) │ 20.16.2 │ 22.5.1 │
├───────────────────┼─────────┼────────┤
│ eslint (dev)      │ 8.57.0  │ 9.9.1  │
├───────────────────┼─────────┼────────┤
│ jsdom (dev)       │ 24.1.3  │ 25.0.0 │
├───────────────────┼─────────┼────────┤
│ vitest (dev)      │ 1.6.0   │ 2.0.5  │
└───────────────────┴─────────┴────────┘
➜  vue-project pnpm run type-check       

> vue-project@0.0.0 type-check /Users/yanhao2023/Downloads/vue-project
> vue-tsc --build --force

src/components/HelloWorld.vue:42:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

42 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/App.vue:86:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

86 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/WelcomeItem.vue:88:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

88 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconDocumentation.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

8 /// <reference types="vue/__globalTypes_3.4_false" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconTooling.vue:20:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

20 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconEcosystem.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

8 /// <reference types="vue/__globalTypes_3.4_false" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconCommunity.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

8 /// <reference types="vue/__globalTypes_3.4_false" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconSupport.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

8 /// <reference types="vue/__globalTypes_3.4_false" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/TheWelcome.vue:89:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

89 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/views/HomeView.vue:10:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

10 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/views/AboutView.vue:16:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

16 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/HelloWorld.vue:42:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

42 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/App.vue:86:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

86 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/WelcomeItem.vue:88:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

88 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconDocumentation.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

8 /// <reference types="vue/__globalTypes_3.4_false" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconTooling.vue:20:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

20 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconEcosystem.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

8 /// <reference types="vue/__globalTypes_3.4_false" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconCommunity.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

8 /// <reference types="vue/__globalTypes_3.4_false" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconSupport.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

8 /// <reference types="vue/__globalTypes_3.4_false" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/TheWelcome.vue:89:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

89 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/views/HomeView.vue:10:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

10 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/views/AboutView.vue:16:23 - error TS2688: Cannot find type definition file for 'vue/__globaltypes_3.4_false'.

16 /// <reference types="vue/__globalTypes_3.4_false" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 22 errors.

 ELIFECYCLE  Command failed with exit code 2.
➜  vue-project 

```

--- a/packages/language-core/lib/codegen/script/index.ts
+++ b/packages/language-core/lib/codegen/script/index.ts
@@ -51,7 +51,7 @@ export interface ScriptCodegenOptions {
 export function* generateScript(options: ScriptCodegenOptions): Generator<Code, ScriptCodegenContext> {
 	const ctx = createScriptCodegenContext(options);
 
-	yield `/// <reference types="${options.vueCompilerOptions.lib}/__globalTypes_${options.vueCompilerOptions.target}_${options.vueCompilerOptions.strictTemplates}" />${newLine}`;
+	yield `/// <reference types="${options.vueCompilerOptions.lib}/dist/__globalTypes_${options.vueCompilerOptions.target}_${options.vueCompilerOptions.strictTemplates}.d.ts" />${newLine}`;
 
 	if (options.sfc.script?.src) {
 		yield* generateSrc(options.sfc.script, options.sfc.script.src);

--- a/packages/tsc/index.ts
+++ b/packages/tsc/index.ts
@@ -27,7 +27,7 @@ export function run(tscPath = require.resolve('typescript/lib/tsc')) {
 						: options.host?.getCurrentDirectory() ?? ts.sys.getCurrentDirectory();
 					const libDir = require.resolve(`${vueOptions.lib}/package.json`, { paths: [rootDir] })
 						.slice(0, -'package.json'.length);
-					const globalTypesPath = `${libDir}__globalTypes_${vueOptions.target}_${vueOptions.strictTemplates}.d.ts`;
+					const globalTypesPath = `${libDir}dist/__globalTypes_${vueOptions.target}_${vueOptions.strictTemplates}.d.ts`;
 					const globalTypesContents = vue.generateGlobalTypes(vueOptions.lib, vueOptions.target, vueOptions.strictTemplates);
 					ts.sys.writeFile(globalTypesPath, globalTypesContents);
 				} catch { }

--- a/packages/tsc/tests/dts.spec.ts
+++ b/packages/tsc/tests/dts.spec.ts
@@ -37,7 +37,7 @@ describe('vue-tsc-dts', () => {
 				: options.host?.getCurrentDirectory() ?? ts.sys.getCurrentDirectory();
 			const libDir = require.resolve(`${vueOptions.lib}/package.json`, { paths: [rootDir] })
 				.slice(0, -'package.json'.length);
-			const globalTypesPath = `${libDir}__globalTypes_${vueOptions.target}_${vueOptions.strictTemplates}.d.ts`;
+			const globalTypesPath = `${libDir}dist/__globalTypes_${vueOptions.target}_${vueOptions.strictTemplates}.d.ts`;
 			const globalTypesContents = vue.generateGlobalTypes(vueOptions.lib, vueOptions.target, vueOptions.strictTemplates);
 			ts.sys.writeFile(globalTypesPath, globalTypesContents);
 		} catch { }

--- a/packages/typescript-plugin/index.ts
+++ b/packages/typescript-plugin/index.ts
@@ -63,7 +63,7 @@ const plugin: ts.server.PluginModuleFactory = mods => {
 				try {
 					const libDir = require.resolve(`${options.lib}/package.json`, { paths: [proj.getCurrentDirectory()] })
 						.slice(0, -'package.json'.length);
-					const globalTypesPath = `${libDir}__globalTypes_${options.target}_${options.strictTemplates}.d.ts`;
+					const globalTypesPath = `${libDir}dist/__globalTypes_${options.target}_${options.strictTemplates}.d.ts`;
 					const globalTypesContents = vue.generateGlobalTypes(options.lib, options.target, options.strictTemplates);
 					proj.writeFile(globalTypesPath, globalTypesContents);
 				} catch { }

